pymc/model.py

```diff
@@ -788,6 +856,54 @@ def logp_elemwiset(
 
         return logp_elemwise
 
+    def dlogpt(
+        self,
+        vars: Optional[Union[Variable, List[Variable]]] = None,
+        jacobian: bool = True,
+    ) -> List[Variable]:
+        if vars is None:
+            value_vars = None
+        else:
+            if not isinstance(vars, (list, tuple)):
+                vars = [vars]
+
+            value_vars = []
+            for i, var in enumerate(vars):
+                value_var = self.rvs_to_values.get(var)
+                if value_var is not None:
+                    value_vars.append(value_var)
+                else:
+                    raise ValueError(
+                        f"Requested variable {var} not found among the model variables"
+                    )
+
+        cost = self.logpt if jacobian else self.logp_nojact
+        return gradient(cost, value_vars)
+
+    def d2logpt(
+        self,
+        vars: Optional[Union[Variable, List[Variable]]] = None,
+        jacobian: bool = True,
+    ) -> List[Variable]:
+        if vars is None:
+            value_vars = None
+        else:
+            if not isinstance(vars, (list, tuple)):
+                vars = [vars]
+
+            value_vars = []
+            for i, var in enumerate(vars):
+                value_var = self.rvs_to_values.get(var)
+                if value_var is not None:
+                    value_vars.append(value_var)
+                else:
+                    raise ValueError(
+                        f"Requested variable {var} not found among the model variables"
+                    )
+
+        cost = self.logpt if not jacobian else self.logp_nojact
+        return hessian(cost, value_vars)
+
     @property
     def logpt(self):
```

Comment:
```suggestion
    def logpt(self) -> Variable:
```
I like the refactor. Adding more type hints will make it easier for beginners like me who can't remember which property is a variable and which one is a function...